plugins/catalog-backend-module-bitbucket-server/api-report.md

```diff
@@ -0,0 +1,97 @@
+## API Report File for "@backstage/plugin-catalog-backend-module-bitbucket-server"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+import { BitbucketServerIntegrationConfig } from '@backstage/integration';
+import { Config } from '@backstage/config';
+import { Entity } from '@backstage/catalog-model';
+import { EntityProvider } from '@backstage/plugin-catalog-backend';
+import { EntityProviderConnection } from '@backstage/plugin-catalog-backend';
+import { LocationSpec } from '@backstage/plugin-catalog-backend';
+import { Logger } from 'winston';
+import { TaskRunner } from '@backstage/backend-tasks';
+
+// @public
+export type BitbucketLocationParser = (options: {
+  client: BitbucketServerClient;
+  location: LocationSpec;
+  logger: Logger;
+}) => AsyncIterable<Entity>;
+
+// @public (undocumented)
+export type BitbucketProject = {
+  key: string;
+};
+
+// @public (undocumented)
+export type BitbucketRepository = {
+  project: {
+    key: string;
+  };
+  slug: string;
+  description: string;
+  links: Record<
+    string,
+    {
+      href: string;
+    }[]
+  >;
+};
+
+// @public
+export class BitbucketServerClient {
```

Comment:
Do we need to export the client or is this something which is rather internal? Or maybe even separate like for Bitbucket Cloud? (e.g., in case the client should be reused at other modules like scaffolder, etc.)

Reply:
Currently the `BitbucketServerLocationParser` uses the client, which is why it's exported.
I also thought about extracting the client to its own plugin, however I figured that can be done in a future PR if we decide to.